src/nsj_rest_lib/dao/dao_base.py

```diff
@@ -804,6 +805,41 @@ def insert(self, entity: EntityBase, sql_read_only_fields: List[str] = []):
 
         return entity
 
+    def _sql_upsert_fields(
+        self,
+        entity: EntityBase,
+        ignore_nones: bool = False,
+        sql_read_only_fields: List[str] = [],
+    ) -> str:
+        """
+        Retorna lista com os campos para upsert, no padrão "field = excluded.field"
+        """
+
+        # Building SQL fields
+        if ignore_nones:
+            fields = [
+                f"{k} = excluded.{k}"
+                for k in entity.__dict__
+                if not callable(getattr(entity, k, None))
+                and not k.startswith("_")
+                and getattr(entity, k) is not None
```

Comment:
Não é melhor trocar essa linha por:

and (ignore_nones and getattr(entity, k) is not None or not ignore_nones)

E, depois, remover o "if ignore_nones:" ?

Reply:
Poderia remover o else também.